destination/destination_integration_test.go

```diff
@@ -93,143 +74,321 @@ func TestDestination_WriteIntegration(t *testing.T) {
 					"attr_2": false,
 				},
 				"achievements": []string{"achievement_0", "achievement_1"},
-			},
-		})
-		is.NoErr(err)
+			}},
+		},
+		{
+			Operation: sdk.OperationSnapshot,
+			Payload: sdk.Change{After: sdk.StructuredData{
+				"id":        43,
+				"name":      "Nick",
+				"is_active": false,
+				"attributes": map[string]any{
+					"attr_0": "string",
+					"attr_1": 1,
+					"attr_2": false,
+				},
+				"achievements": []string{"achievement_0", "achievement_1"},
+			}},
+		},
+	}
 
-		err = dest.Teardown(ctx)
-		is.NoErr(err)
-	})
+	n, err = dest.Write(ctx, records)
+	is.NoErr(err)
+	is.Equal(n, len(records))
+
+	cancel()
 
-	t.Run("update", func(t *testing.T) {
-		err = dest.Open(ctx)
+	err = dest.Teardown(context.Background())
+	is.NoErr(err)
+}
+
+func TestDestination_Write_Update(t *testing.T) {
+	t.Parallel()
+
+	var (
+		ctx = context.Background()
+		cfg = prepareConfig(t)
+		is  = is.New(t)
+	)
+
+	repo, err := repository.New(cfg[models.ConfigURL])
+	is.NoErr(err)
+	defer repo.Close()
+
+	err = createTable(repo, cfg[models.ConfigTable])
+	is.NoErr(err)
+
+	defer func() {
+		err = dropTable(repo, cfg[models.ConfigTable])
 		is.NoErr(err)
+	}()
+
+	err = insertData(repo, cfg[models.ConfigTable])
+	is.NoErr(err)
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	dest := NewDestination()
 
-		err = dest.Write(ctx, sdk.Record{
-			Metadata: map[string]string{
-				"action": "update",
-			},
-			Payload: sdk.StructuredData{
+	err = dest.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = dest.Open(ctx)
+	is.NoErr(err)
+
+	n := 0
+	n, err = dest.Write(ctx, []sdk.Record{
```

Comment:
why not ?
n, err:=dest.Write(ctx, []sdk.Record{

Reply:
Yep, you're right. Changed!